<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>592</width>
    <height>805</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>1</horstretch>
    <verstretch>1</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Tester for Arduino</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_1">
      <property name="sizeConstraint">
       <enum>QLayout::SetMinimumSize</enum>
      </property>
      <item>
       <widget class="QTableView" name="tableView">
        <property name="minimumSize">
         <size>
          <width>283</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>283</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Shows Pins of Arduino</string>
        </property>
        <property name="cornerButtonEnabled">
         <bool>true</bool>
        </property>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_1">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>120</height>
           </size>
          </property>
          <property name="text">
           <string>Connect the buttons of a row below to arduino pins in the edit field on the right.
Buttons in a row are separated by ':'.
More than one pin number per button is separated by ','.
Positive pin number means button pressed = 1.
Negative pin number means button pressed = 0.</string>
          </property>
          <property name="textFormat">
           <enum>Qt::AutoText</enum>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="2" column="3">
           <widget class="QLineEdit" name="editClearDownF2">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Arduino pin numbers for buttons Clear:Down:F2</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btnRight">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Right</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="btnUp">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Up</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLineEdit" name="editRightEnterLeft">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Arduino pin numbers for buttons Right:Enter:Left</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="btnEscape">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>ESC</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btnLeft">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Left</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="btnClear">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btnEnter">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Enter</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="btnDown">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>Down</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLineEdit" name="editEscUpF1">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Arduino pin numbers for buttons ESC:Up:F1.</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="btnF1">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>F1</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QPushButton" name="btnF2">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="text">
             <string>F2</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textLCD_Sim">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>100</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Courier 10 Pitch</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="toolTip">
           <string>simulated LCD display</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="btnSendText">
            <property name="toolTip">
             <string>Sends a text with newline to the simulated Arduinos  serial device</string>
            </property>
            <property name="text">
             <string>Send Line to Serial...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bntSendFile">
            <property name="toolTip">
             <string>Sends a text file with newlines to the simulated Arduinos serial device</string>
            </property>
            <property name="text">
             <string>Send File to Serial</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLineEdit" name="textToSendToSerialDevice">
          <property name="toolTip">
           <string>Input text line for simulated Serial Device</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QPushButton" name="btnClearText">
            <property name="toolTip">
             <string>Clears the output text from the simulated Arduinos serial device</string>
            </property>
            <property name="text">
             <string>Clear Serial Output</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnDoTest">
            <property name="toolTip">
             <string>Starts the Unittests within the Function doTest()</string>
            </property>
            <property name="text">
             <string>Start Unit Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textReceivedFromSerialDevice">
          <property name="toolTip">
           <string>Output text from simulated Serial device</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnLogicAnalyzer">
          <property name="text">
           <string>Logic Analysis</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>592</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
